server/sonar-server/src/main/java/org/sonar/server/measure/ws/MeasureValueFormatter.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * SonarQube :: Server
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+package org.sonar.server.measure.ws;
+
+import org.sonar.api.measures.Metric;
+import org.sonar.db.measure.MeasureDto;
+import org.sonar.db.metric.MetricDto;
+
+class MeasureValueFormatter {
+  private static final double DELTA = 0.000001d;
+
+  private MeasureValueFormatter() {
+    // static methods
+  }
+
+  static String formatMeasureValue(MeasureDto measure, MetricDto metric) {
+    Metric.ValueType metricType = Metric.ValueType.valueOf(metric.getValueType());
+    Double doubleValue = measure.getValue();
+    String stringValue = measure.getData();
+
+    switch (metricType) {
+      case BOOL:
+        return formatBoolean(doubleValue);
+      case INT:
+      case MILLISEC:
+        return formatInteger(doubleValue);
+      case WORK_DUR:
+        return formatLong(doubleValue);
+      case FLOAT:
+      case PERCENT:
+      case RATING:
+        return String.valueOf(doubleValue);
+      case LEVEL:
+      case STRING:
+      case DATA:
+      case DISTRIB:
+        return stringValue;
+      default:
+        throw new IllegalArgumentException("Unsupported metric type: " + metricType.name());
+    }
+  }
+
+  static String formatDoubleValue(Double value, MetricDto metric) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) The Cyclomatic Complexity of this method "formatDoubleValue" is 17 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)